Automattic/woocommerce-payments
# feat: add confirmation modal component

Helps with https://github.com/Automattic/woocommerce-payments/issues/1564

#### Changes proposed in this Pull Request









It looks like we're starting to use a few of these modals within our codebase.
I'd like to add a "confirmation modal" component, which takes care of the layout of the modals we're using in the codebase.

These are all the existing modals:
- https://d.pr/i/YvnEJu
- https://d.pr/i/d5D7Kn

And we're adding also this:
- https://d.pr/i/5l2or7 - which I'm doing for the confirmation dialog task: https://github.com/Automattic/woocommerce-payments/issues/1564

As you can see, they all have a:
- title
- body
- separator
- CTAs

The separator and the CTAs all need to be styled each time. I thought about creating a separate component where the styles can be defined once.
I thought that for the confirmation dialog task ( https://github.com/Automattic/woocommerce-payments/issues/1564 ) I would DRY up the code a bit.

#### Testing instructions





- Ensure you have the latest version of the dev tools: `cd docker/wordpress/wp-content/plugins/woocommerce-payments-dev-tools && git checkout trunk && git pull`
- Ensure you have the "grouped settings" flag enabled in WCPay Dev Tools
- Ensure you have the "Add UPE additional payment methods" flag enabled in WCPay Dev Tools
- Go to your WCPay Home: http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout
- Clicking on "add payment method" or the "delete" button on each payment method displays the modal
- No change in functionality compared to the current `develop` branch

-------------------
